contrib/linearize/linearize-data.py

```diff
@@ -8,23 +8,33 @@
 #
 
 from __future__ import print_function, division
+try: # Python 3
+    import http.client as httplib
+except ImportError: # Python 2
+    import httplib
 import json
 import struct
 import re
 import os
 import os.path
 import base64
-import httplib
 import sys
 import hashlib
 import datetime
 import time
 from collections import namedtuple
+from binascii import hexlify, unhexlify
 
 settings = {}
 
+##### Switch endian-ness #####
+def hex_switchEndian(s):
+	""" Switches the endianness of a hex string (in pairs of hex chars) """
+	pairList = [s[i:i+2].encode() for i in range(0, len(s), 2)]
+	return b''.join(pairList[::-1]).decode()
```

Comment:
The `encode` and `decode` call seem to be unnecessary. Also my highly unscientific test shows that 
```
s.decode('hex')[::-1].encode('hex')
```
is about 5 times faster than the above loop, but may be a bit more confusing.